File: moviejava/app/src/main/java/com/example/moviejava/model/up_coming/UpComing.java
package com.example.moviejava.model.up_coming;


import com.example.moviejava.model.Dates;
import com.example.moviejava.model.Results;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class UpComing {

    @SerializedName("dates")
    public Dates dates;
    @SerializedName("page")
    public int page;
    @SerializedName("results")
    public ArrayList<Results> results;
    @SerializedName("total_pages")
    public int totalPages;
    @SerializedName("total_results")
    public int totalResult;

}

File: movie/app/src/main/java/com/example/movie/model/up_coming/UpComing.kt
package com.example.movie.model.up_coming

import com.example.movie.model.Dates
import com.example.movie.model.Results
import com.google.gson.annotations.SerializedName

data class UpComing(

    @SerializedName("dates") var dates: Dates? = Dates(),
    @SerializedName("page") var page: Int? = null,
    @SerializedName("results") var results: ArrayList<Results> = arrayListOf(),
    @SerializedName("total_pages") var totalPages: Int? = null,
    @SerializedName("total_results") var totalResults: Int? = null

)